lib/grammers-tl-gen/src/structs.rs

```diff
@@ -60,6 +60,33 @@ fn write_struct<W: Write>(
     _metadata: &Metadata,
     config: &Config,
 ) -> io::Result<()> {
+    if let Some(docs) = &def.docs {
+        writeln!(
+            file,
+            "{}/// {}",
+            indent,
+            docs.documentation.description.replace("\n", "<br>")
+        )?;
+
+        writeln!(
+            file,
+            r#"{}/// generated based on <a href="{link}">{name}</a>"#,
+            indent,
+            name = docs.name,
+            link = docs.url_path
+        )?;
+        writeln!(file, "{}///", indent,)?;
+
+        if docs.documentation.errors.len() > 0 {
+            writeln!(file, "{}/// possible errors (these are not all the possible errors, and some may no longer apply):<br>", indent)?;
+
+            writeln!(
+                    file,
+                    "{}/// <table><thead><tr><th>error</th><th>code</th><th>description</th></tr></thead>{}</table>",
+                    indent, docs.documentation.errors.iter().map(|(name,error)|format!("<tr><td>{}</td><td>{}</td><td>{}</td></tr>",name,error.code,error.description)).collect::<Vec<_>>().join(" ")
```

Comment:
Is this the best `cargo fmt` can do here? It's a bit hard to read.

Reply:
I don't think `rustfmt` does much inside macros; you have to manually format function-like macros like functions yourself. (Changing `writeln!(` to `writeln(`, formatting, and then changing `writeln(` back to `writeln!(` works.)

Reply:
I thought it would have a special-case for macros like `write(ln)` and `print(ln)`; maybe the line is too long / complicated, I remember cases of those failing to `fmt` too.

In either case I'd like to see this formatted a bit better.